packages/ERTP/src/eventualWithdraw.js

```diff
@@ -0,0 +1,40 @@
+// @ts-check
+import { E } from '@agoric/eventual-send';
+import { makePromiseKit } from '@agoric/promise-kit';
+
+import './types';
+
+/** @type {EventualWithdraw} */
+export const eventualWithdraw = (
+  purse,
+  amountMath,
+  searchAmount,
+  iterations = 1,
+) => {
+  const notifier = E(purse).getCurrentAmountNotifier();
+  const paymentPromiseKit = makePromiseKit();
+
+  const tryToFind = async lastUpdateCount => {
+    if (iterations <= 0) {
+      paymentPromiseKit.reject(Error('searchAmount could not be withdrawn'));
+      return;
+    }
+    const { value: currentAmount, updateCount } = await E(
+      notifier,
+    ).getUpdateSince(lastUpdateCount);
+    const foundAmount = amountMath.find(currentAmount, searchAmount);
```

Comment:
See https://github.com/Agoric/agoric-sdk/pull/2155#pullrequestreview-560256142

I don't understand what this line means in a way that generalizes over the different kinds of erights.